editoast/src/views/path/properties.rs

```diff
@@ -175,15 +244,42 @@ async fn post(
 ) -> Result<Json<PathProperties>> {
     // Extract information from parameters
     let conn = &mut db_pool.get().await?;
-    let infra_version = retrieve_infra_version(conn, infra_id).await?;
     let query_props: Properties = props.into();
     let mut valkey_conn = valkey.get_connection().await?;
+    let infra = Infra::retrieve_or_fail(conn, infra_id, || PathfindingError::InfraNotFound {
+        infra_id,
+    })
+    .await?;
+
+    // Get track section offsets.
+    let track_section_offsets = path_properties_input.find_track_section_offsets();
+
+    // Get the IDs of the track sections.
+    let track_section_ids = track_section_offsets
+        .iter()
+        .map(|ts| ts.0.clone()) // Extract track section IDs.
+        .collect();
+
+    // Fetch the track section objects using the IDs.
+    let track_sections = infra
+        .get_objects(conn, ObjectType::TrackSection, &track_section_ids)
+        .await?
+        .into_iter()
+        .map(|track_section| track_section.railjson)
+        .map(serde_json::from_value::<TrackSection>)
+        .collect::<Result<Vec<_>, serde_json::Error>>()?;
+
+    let path_item_position_ratio = track_sections
+        .iter()
+        .map(|ts| path_properties_input.get_ratio(ts))
+        .flatten()
+        .collect::<Vec<(String, f64)>>();
```

Comment:
`track_sections` is not used afterwards, so I would collapse the 2 chain of iterators into one to avoid doing 2 `.collect()`, what do you think?